src/app/sub/mount_point.rs

```diff
@@ -0,0 +1,41 @@
+use std::{ffi::OsStr, path::Path};
+
+pub fn find_mount_point<'a>(
+    path: &Path,
+    mount_points: impl IntoIterator<Item = &'a Path>,
+) -> Option<&'a Path> {
+    mount_points
+        .into_iter()
+        .filter(|mnt| path.starts_with(&*mnt.to_string_lossy()))
```

Comment:
Why is `to_string_lossy` here? `mnt` is a `&Path`, it should fit [`Path::starts_with`](https://doc.rust-lang.org/std/path/struct.Path.html#method.starts_with) just fine.

```suggestion
        .filter(|mnt| path.starts_with(mnt))
```